retail-managment-system/rms-ui/src/app/pages/transaction/transaction.component.html

```diff
@@ -0,0 +1,112 @@
+    </table>
+    <form class="col-lg-4 p-3" [formGroup]="transactionPostForm">
+      <div class="mat-typography d-flex justify-content-between px-2">
+        <h2 class="mat-headline">{{
+                transactionPostForm.controls["id"].value
+                  ? ("update" | translate)
+                  : ("add" | translate)
+              }}
+
+          {{ "transactions.transaction" | translate }}</h2>
+        <mat-icon class="icon-add" (click)="restartForm(); isButtonVisible = true" [hidden]="isButtonVisible">add</mat-icon>
+      </div>
+      <mat-form-field class="row" appearance="outline">
+        <mat-label>{{"transactions.amount"|translate}}</mat-label>
+        <input type="number" matInput required formControlName="amount">
+        <mat-error *ngIf="transactionPostForm.controls['amount'].hasError('min')">
+          {{ "transactions.amount-validation" | translate
+            }}<strong> {{ "transactions.required" | translate }}</strong>
+        </mat-error>
+        <mat-error *ngIf="transactionPostForm.controls['amount'].hasError('required')">
+          <strong> {{ "transactions.required" | translate }}</strong>
+        </mat-error>
+      </mat-form-field>
+      <mat-form-field class="row" appearance="outline">
+        <mat-label>{{"transactions.date"|translate}}</mat-label>
+        <input matInput required type="date" formControlName="date">
+        <mat-error *ngIf="transactionPostForm.controls['date'].hasError('required')">
+          <strong> {{ "transactions.required" | translate }}</strong>
+        </mat-error>
+      </mat-form-field>
+      <mat-form-field class="row" appearance="outline">
+        <mat-label>{{"transactions.type"|translate}}</mat-label>
+        <mat-select required formControlName="transactionType">
+          <mat-option [value]="type" *ngFor="let type of transactionTypeEnum">{{type}}</mat-option>
+        </mat-select>
+        <mat-error *ngIf="transactionPostForm.controls['transactionType'].hasError('required')">
+          <strong> {{ "transactions.required" | translate }}</strong>
+        </mat-error>
+      </mat-form-field>
+      <mat-form-field class="row" appearance="outline">
+        <mat-label>{{"transactions.investor"|translate}}</mat-label>
+        <mat-select required formControlName="investor" [compareWith]="compareFn">
+          <mat-option *ngFor="let type of investorNames" [value]="type">{{type.fullName}}</mat-option>
+        </mat-select>
+        <mat-error *ngIf="transactionPostForm.controls['investor'].hasError('required')">
+          <strong> {{ "transactions.required" | translate }}</strong>
+        </mat-error>
+      </mat-form-field>
+      <div class="px-3 d-flex justify-content-end">
+        <button class="mx-1" mat-raised-button color="blue" type="submit" (click)="onSubmit()"
+          [disabled]="transactionPostForm.invalid">{{transactionPostForm.controls["id"].value
```

Comment:
make button disabled in same time like other pages

Reply:
done